api/desecapi/authentication.py

```diff
@@ -127,18 +126,10 @@ def authenticate(self, request):
             msg = 'No password URL parameter provided.'
             raise exceptions.AuthenticationFailed(msg)
 
-        return self.authenticate_credentials(request.query_params['username'], request.query_params['password'])
-
-    def authenticate_credentials(self, _, key):
         try:
-            user, token = TokenAuthentication().authenticate_credentials(key)
-        except self.model.DoesNotExist:
-            raise exceptions.AuthenticationFailed('badauth')
-
-        if not user.is_active:
-            raise exceptions.AuthenticationFailed('badauth')
-
-        return token.user, token
+            return self.authenticate_credentials(request.query_params['username'], request.query_params['password'])
+        except Exception:
```

Comment:
this may be too broad and swallow exceptions that we would want to know about (think, e.g., a lost db connection)

Reply:
I was thinking to catch both exceptions from `authenticate_credentials()` (that would be `AuthenticationFailed`), but also from the inner call to `get_domain_for_qname()`, which can raise a Django validation error (via `clean_fields`). I wasn't sure how to do it smoothly without the authentication code being aware of the latter.

Maybe we should catch the validation error in `get_domain_for_qname()` and turn it into a `ValueError`, then catch that here?

Reply:
> Maybe we should catch the validation error in get_domain_for_qname() and turn it into a ValueError, then catch that here?

For values that fail validation I suggest that `get_domain_for_qname()` returns either None or raises an "not found" exception, which should be converted by `authenticate_credentials()` into an authentication failure

Reply:
I turned `get_domain_for_qname` into `filter_qname` which is a bit more generic. It now raises a ValueError if a syntactically invalid domain name is passed. We now catch that during authentication and in the dynDNS view.